File: python/twopointer/ValidPalindrome.py
import string


class ValidPalindrome:
    def isPalindrome(self, str: string):
        left = 0
        right = len(str)-1

        while left < right:
            if str[left] != str[right]:
                return False
            left += 1
            right -= 1

        return True;

vp = ValidPalindrome()
print(vp.isPalindrome('abba'))

File: java/src/prep/leetcode/ValidPalindrome.java
package prep.leetcode;

public class ValidPalindrome {
	public boolean isPalindrome(String s) 
	{
		StringBuffer sb = new StringBuffer();
		for(char c : s.toCharArray())
		{
			if(Character.isLetter(c) || Character.isDigit(c))
				sb.append(Character.toLowerCase(c));
		}
			
		
		int left=0;
		int right=sb.length()-1;
		
		while(left<right)
		{
			
			if(sb.charAt(right)!=sb.charAt(left))
				return false;
			
			left++;
			right--;
			
		}
		return true;
    }

	public static void main(String[] args) {
		ValidPalindrome vp = new ValidPalindrome();
		System.out.println(vp.isPalindrome("A man, a plan, a canal: Panama"));

	}

}
